### ISSUE TITLE
Please support superscripts and subscripts before the base symbol?

### ISSUE THREAD

**Issue by ndclause** [typst/typst#690]
Hi, I love Typst!

I wish it can support "prescripts", i.e. superscripts and subscripts before the base symbol. Notably, the upper and lower prescripts need to be aligned horizontally on the right edge, not on the left.

The notation is not rare. For instance, isotopes in chemistry are denoted using prescripts, like the most common oxygen 
<img width="36" alt="Screenshot 2023-04-09 at 13 44 08" src="https://user-images.githubusercontent.com/104699533/230788608-3e11c26e-33b7-495c-9dc4-491b5862631f.png">, and isotope notations are needed by chemists, physicists, biologists, geologists, archaeologist, etc. It is also used to denote [combination/permutation (math)](https://www.pinterest.com/pin/basic-statistics-probability-formulas-pdf-download--311592867980917008/), [tetration (math)](https://en.wikipedia.org/wiki/Tetration), [accidentals (music)](https://en.wikipedia.org/wiki/List_of_musical_symbols), etc.. These usages are supported in Latex's famous `amsmath`, and also niche packages like `leftidx`.

I did spent quite some time in trying to make the isotope example work with existing Typst feature with e.g. `hide` or `table`, but the effect is slightly off, and doesn't work well with the surrounding text. Anyway, I really look forward to a clean and clear native solution, e.g. `attach(O, pre_top: 16, pre_bottom: 8)` for the oxygen isotope.

**Comment by Leedehai** [comment]
Hi @laurmaedje I noticed you just added a "feature request" label. Wondering if there are any thoughts on the PR #699? Thanks.

**Comment by laurmaedje** [comment]
@Leedehai Yeah, I just label everything :) I've seen your PR and will review it soon.
